box-content-sdk/src/main/java/com/box/androidsdk/content/requests/MultiputResponseHandler.java

```diff
@@ -0,0 +1,50 @@
+package com.box.androidsdk.content.requests;
+
+import com.box.androidsdk.content.BoxException;
+import com.box.androidsdk.content.models.BoxIteratorItems;
+
+import java.net.HttpURLConnection;
+
+/**
+ * A request handler that is designed to handle the parsing logic necessary for a BoxRequestDownload.
+ */
+
+public class MultiputResponseHandler extends BoxRequest.BoxRequestHandler<BoxRequestsFile.CommitUploadSession> {
+    /**
+     * A request handler that is designed to handle the parsing logic necessary for a BoxRequestDownload.
+     */
+
+        protected static final int DEFAULT_NUM_RETRIES = 2;
+        protected static final int DEFAULT_MAX_WAIT_MILLIS = 90 * 1000;
+
+        protected int mNumAcceptedRetries = 0;
+        protected int mRetryAfterMillis = 1000;
+
+    public MultiputResponseHandler(BoxRequestsFile.CommitUploadSession request) {
+        super(request);
+    }
+
+
+    @Override
+        public BoxIteratorItems onResponse(Class clazz, BoxHttpResponse response) throws IllegalAccessException, InstantiationException, BoxException {
+            if (response.getResponseCode() == HttpURLConnection.HTTP_ACCEPTED) {
+                try {
+                    // First attempt to use Retry-After header, all failures will eventually fall back to exponential backoff
+                    if (mNumAcceptedRetries < DEFAULT_NUM_RETRIES) {
+                        mNumAcceptedRetries++;
+                        mRetryAfterMillis = getRetryAfterFromResponse(response, 1);
+                    } else if (mRetryAfterMillis < DEFAULT_MAX_WAIT_MILLIS) {
+                        // Exponential back off with some randomness to avoid traffic spikes to server
+                        mRetryAfterMillis *= (1.5 + Math.random());
+                    } else {
+                        // Give up after the maximum retry time is exceeded.
+                        throw new BoxException.MaxAttemptsExceeded("Max wait time exceeded.", mNumAcceptedRetries);
+                    }
+                    Thread.sleep(mRetryAfterMillis);
+                    return (BoxIteratorItems) mRequest.send();
+                } catch (InterruptedException e) {
+                    throw new BoxException(e.getMessage(), response);
+                }
+            } else return (BoxIteratorItems) super.onResponse(clazz, response);
```

Comment:
Since we are creating a UploadSession per file, isn't the response supposed to be of the type BoxFile?

Reply:
The api says it returns files, but when we tested it seems to be a list of files.
This is what Kartik said "The commit response format depends on whether you use API 2.0 or 2.1.  The commit response is identical to the response format of a single upload request for each respective version (i.e. API 2.0 commit returns same format as API 2.0 regular upload). " I can follow up, but for now, we should use this till we try out 2.1 I guess